libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -185,6 +190,40 @@ queryDRepStakeDistr nes creds
   where
     distr = psDRepDistr . fst $ finishedPulserState nes
 
+queryDRepStakeDistrEfficient ::
+  ConwayEraGov era =>
+  NewEpochState era ->
+  Set (DRep (EraCrypto era)) ->
+  Map (DRep (EraCrypto era)) Coin
+queryDRepStakeDistrEfficient nes creds = Map.map fromCompact $ Map.foldlWithKey' computeDistr mempty selectedDReps
+  where
+    -- create a map of registered dreps with the abstain and noconf dreps included
+    -- filter this map if a set of dreps was provided for querying
+    selectedDReps
+      -- TODO: add noconf and abstain dreps to maps
```

Comment:
How should I go about this? I really don't want to filter the `dRepMap` (extracted from `UMap`) for the `AlwaysAbstain` and `NoConfidence` DReps, get their staking credentials and lookup the delegated rewards. Is there another way to get the stake delegated to `NoConfidence` and `AlwaysAbstain` DReps and is there a way to know their staking credentials without filtering the `dRepMap`?